glean-core/android/src/main/java/mozilla/telemetry/glean/private/EventMetricType.kt

```diff
@@ -54,6 +54,21 @@ enum class NoExtraKeys(
     // deliberately empty
 }
 
+/**
+ * A class that can be converted into key-value pairs of event extras.
+ * This will be automatically implemented for event properties of an [EventMetricType].
+ */
+interface EventExtras {
+    /**
+     * Convert the event extras into 2 lists:
+     *
+     * 1. The list of extra key indices.
+     *    Unset keys will be skipped.
+     * 2. The list of extra values.
+     */
+    fun toFfiExtra(): Pair<IntArray, StringArray>
+}
+
 /**
  * This implements the developer facing API for recording events.
  *
```

Comment:
Leaving the comment here since I can't comment on the specific line. Can we add a [Deprecated](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-deprecated/) annotation on line 122 ` fun record(extra: Map<ExtraKeysEnum, String>? = null) {`?